unity-renderer/Assets/UIComponents/Tests/SectionSelectorComponentViewTests.cs

```diff
@@ -0,0 +1,145 @@
+
+        // Act
+        sectionSelectorComponent.SetSections(testSections);
+
+        // Assert
+        Assert.AreEqual(testSections, sectionSelectorComponent.model.sections, "The section list does not match in the model.");
+        Assert.AreEqual(testSections.Count + 1, sectionSelectorComponent.transform.childCount, "The number of instantiated sections does not match.");
+    }
+
+    [Test]
+    public void GetSectionCorrectly()
+    {
+        // Arrange
+        List<SectionToggleModel> testSections = CreateTestSections(2);
+        sectionSelectorComponent.SetSections(testSections);
+
+        // Act
+        ISectionToggle existingSection1 = sectionSelectorComponent.GetSection(0);
+        ISectionToggle existingSection2 = sectionSelectorComponent.GetSection(1);
+
+        // Assert
+        Assert.AreEqual(testSections[0].title, existingSection1.GetInfo().title, "The item 1 gotten does not match.");
+        Assert.AreEqual(testSections[1].title, existingSection2.GetInfo().title, "The item 2 gotten does not match.");
+    }
+
+    [Test]
+    public void GetAllSectionsCorrectly()
+    {
+        // Arrange
+        List<SectionToggleModel> testSections = CreateTestSections(2);
+        sectionSelectorComponent.SetSections(testSections);
+
+        // Act
+        List<ISectionToggle> allExistingItems = sectionSelectorComponent.GetAllSections();
+
+        // Assert
+        Assert.AreEqual(testSections[0].title, allExistingItems[0].GetInfo().title, "The section 1 gotten does not match.");
+        Assert.AreEqual(testSections[1].title, allExistingItems[1].GetInfo().title, "The section 2 gotten does not match.");
+        Assert.AreEqual(testSections.Count, allExistingItems.Count, "The number of sections gotten do not match.");
+    }
+
+    [UnityTest]
+    public IEnumerator RemoveAllInstantiatedSectionsCorrectly()
+    {
+        // Arrange
+        List<SectionToggleModel> testSections = CreateTestSections(2);
+        sectionSelectorComponent.SetSections(testSections);
+
+        // Act
+        sectionSelectorComponent.RemoveAllIntantiatedSections();
+        yield return null;
+
+        // Assert
+        Assert.AreEqual(1, sectionSelectorComponent.transform.childCount, "The number of sections does not match.");
+    }
+
+    private SectionSelectorComponentModel CreateTestModel(int numberOfSections)
+    {
+        SectionSelectorComponentModel testModel = new SectionSelectorComponentModel
+        {
+            sections = CreateTestSections(numberOfSections)
+        };
+
+        return testModel;
+    }
+
+    private List<SectionToggleModel> CreateTestSections(int numberOfSections)
+    {
+        List<SectionToggleModel> sections = new List<SectionToggleModel>();
+
+        for (int i = 0; i < numberOfSections; i++)
+        {
+            sections.Add(new SectionToggleModel
+            {
+                icon = Sprite.Create(new Texture2D(10, 10), new Rect(), Vector2.zero),
```

Comment:
Please remember that the `Texture2D` objects have to be destroyed when the suite ends.

Reply:
Good catch! Done in: https://github.com/decentraland/unity-renderer/pull/1175/commits/531831692da512ed2d0095f4e2c0d08be70ee7be